### PR TITLE
Added support for Websockets

### PR DESCRIPTION
**Pull request by jacobm-splunk**
This MR aims to add support for proxying websockets to wiretap.

It skips validation on this endpoint as OpenAPI does not really support websocket specification.

I am not the most familiar with gorilla/websocket so I may have missed something.

### CODE REVIEW

**Comment by daveshanley** [comment]
Amazing work, will review ASAP!

File: cmd/root_command.go [hunk 1]
```diff
@@ -706,11 +715,20 @@ func printLoadedIgnoreRedirectPaths(ignoreRedirects []string) {
...
 		pterm.Printf("🐵 Paths matching '%s' will always follow redirects, regardless of ignoreRedirect settings\n", pterm.LightCyan(x))
 	}
 	pterm.Println()
 }
+
+func printLoadedWebsockets(websockets map[string]*shared.WiretapWebsocketConfig) {
+	pterm.Info.Printf("Loaded %d %s: \n", len(websockets), shared.Pluralize(len(websockets), "websocket", "websockets"))
+
+	for websocket := range websockets {
+		pterm.Printf("🔌 Paths prefixed '%s' will be managed as a websocket\n", pterm.LightCyan(websocket))
```
Comment (daveshanley) [inline cmd/root_command.go:731]: Loving the choice of emojis here.

Comment (jacobm-splunk) [reply cmd/root_command.go:731]: Gotta have emojis

**Comment by jacobm-splunk** [comment]
> A couple of nits, other than that - it looks great.
> 
> Didn't we discuss the need to add in the endpoints in which to forward, or are you simple just doing a straight passthrough for any endpoint?

I have a user add the configuration for their endpoints and then register a handler on each one of those:

```
// Handle Websockets
for websocket := range wiretapConfig.WebsocketConfigs {
	mux.HandleFunc(websocket, handleWebsocket)
}
```
If that's not doing what I expect it to do, then I can totally change it, but since the websocket is opened through a http request to a specific endpoint, then I think this should work?

File: daemon/handle_request.go [hunk 2]
```diff
@@ -238,8 +215,210 @@ func (ws *WiretapService) handleHttpRequest(request *model.Request) {
...
+		Auth:          auth,
+		Variables:     config.CompiledVariables,
+	})
+
+	// Open a new websocket connection with the server
+	dialer := *websocket.DefaultDialer
+	dialer.TLSClientConfig = &tls.Config{InsecureSkipVerify: !*websocketConfig.VerifyCert}
+	serverConn, _, err := dialer.Dial(newRequest.URL.String(), newRequest.Header)
+	if err != nil {
+		ws.config.Logger.Error("Unable to create server connection")
```
[the remark this answers is not in the document]
Comment (jacobm-splunk) [reply daemon/handle_request.go:301]: Copied this verbatim, and added the actual error text

**Pull request merged**